test/e2e/workflow_template_test.go

```diff
@@ -70,6 +70,22 @@ spec:
 
 }
 
+func (s *WorkflowTemplateSuite) TestSubmitWorkflowTemplateWithEnum() {
+	s.Given().
+		WorkflowTemplate("@testdata/workflow-template-with-enum-values.yaml").
+		WorkflowName("my-wf-with-enum").
+		When().
+		CreateWorkflowTemplates().
+		RunCli([]string{"submit", "--from", "workflowtemplate/workflow-template-with-enum-values", "--name", "my-wf-with-enum", "-l", "argo-e2e=true"}, func(t *testing.T, output string, err error) {
+			assert.NoError(t, err)
+		}).
+		WaitForWorkflow().
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
+			assert.Equal(t, status.Phase, v1alpha1.NodeSucceeded)
```

Comment:
don't we want to test failure here?

Reply:
No, `workflow-template-with-enum-values.yaml` contains a valid workflow
It should not result in an error

Reply:
@alexec Should I add a separate failing test

Reply:
I think if you can cover it with unit tests - we do not need an e2e test.